central/resourcecollection/datastore/datastore_impl.go

```diff
@@ -123,6 +127,115 @@ func (ds *datastoreImpl) addCollectionToGraph(obj *storage.ResourceCollection) e
 	return nil
 }
 
+// graphEdgeUpdates stores added and removed edges from the DAG graph
+type graphEdgeUpdates struct {
+	srcID   string
+	added   []string
+	removed []string
+}
+
+func (ds *datastoreImpl) updateCollectionInGraph(obj *storage.ResourceCollection) (*graphEdgeUpdates, error) {
+	ds.initGraphOnce()
+	ds.graphLock.Lock()
+	defer ds.graphLock.Unlock()
+
+	if obj == nil {
+		return nil, errors.New("passed collection must not be nil")
+	}
+
+	// return object struct
+	ret := &graphEdgeUpdates{
+		obj.GetId(),
+		make([]string, 0),
+		make([]string, 0),
+	}
+
+	// get edges for the current object
+	curEdges, err := ds.graph.GetDescendants(obj.GetId())
+	if err != nil {
+		return nil, errors.Wrap(err, "update attempt on unknown collection")
```

Comment:
The error message could be clearer. Maybe `failed to read collection  {id x} from the graph`?

Reply:
I think the context of the id failed to read will be provided by the DAG layer and since this is an error wrap I just want it to provide the context of being a failed update attempt